Project.toml

```diff
@@ -26,12 +26,24 @@ SpecialFunctions = "276daf66-3868-5448-9aa4-cd146d93841b"
 Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 ZygoteRules = "700de1a5-db45-46bc-99cf-38207098b444"
 
+[weakdeps]
+Colors = "5ae59095-9a9b-59fe-a467-6f913c188581"
+Distances = "b4f34e82-e78d-54a5-968a-f98e89d6e8f7"
```

Comment:
I wonder if we should move forward with https://github.com/JuliaStats/Distances.jl/pull/246 instead? But I guess there's also no harm in keeping this code in an extension for a few releases.

Reply:
Doesn't matter from my p.o.v.

Reply:
Using `ZygoteRuleConfig` for `rrules` defined here would prevent any overwrites. That said, it would be much nicer if we didn't maintain 2 separate copies of these rules. The question there is whether Distances.jl would be willing to use Requires or hard deps to support Julia <1.9.

Reply:
BTW there's another difference, I just re-read my comment in the Distances PR and apparently the rules there differ from the currently implemented rules in Zygote:

> The implementation in this PR is different from the code in https://github.com/FluxML/Zygote.jl/pull/923 since tests fail with the rules in the Zygote PR (e.g., due to type inference issues).
The rules in this PR are also all explicit and do not call back into the AD system, in contrast to the current implementation in Zygote.